### PR TITLE
UNR-1616: Factoring Linux worker scripts out of Build.cs

### PR DESCRIPTION
**Pull request by davedolben**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Pulls the Linux worker scripts out of Build.cs and into a common library, and adds a standalone C# executable to write Linux build scripts for use outside of Build.exe (e.g. in a custom build pipeline).

#### Release note
REQUIRED: Add a release note to the `##Unreleased` section of CHANGELOG.md. You can find guidance for writing useful release notes [here](../SpatialGDK/Extras/internal-documentation/how-to-write-good-release-notes.md). Documentation changes are exempt from this requirement.

#### Tests
How did you test these changes prior to submitting this pull request?
* Manually verified that both WriteLinuxScript.exe and Build.exe properly write Linux worker build scripts.
* Manually tested a cloud deployment with a Linux worker launch script written by `Build.exe`
* Manually tested a cloud deployment with a Linux worker launch script written by `WriteLinuxScript.exe`
* Manually tested that simulated players are built with the desired `StartWorker.sh` script. Did not actually test that they run.

What automated tests are included in this PR?
* None

STRONGLY SUGGESTED: How can this be verified by QA?
* Launching a cloud deployment built using Build.exe should be enough to verify this.

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.
@joshuahuburn

### CODE REVIEW

File: SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Common/LinuxScripts.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+using System.IO;
```
Comment (m-samiec) [inline SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Common/LinuxScripts.cs]: Driveby - we should probably have copyrights at the top of these files -
`// Copyright (c) Improbable Worlds Ltd, All Rights Reserved`

Comment (m-samiec) [inline SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Common/LinuxScripts.cs]: Driveby - we should probably have copyrights at the top of these files -
`// Copyright (c) Improbable Worlds Ltd, All Rights Reserved`

**Comment by m-samiec** [comment]
Please increment the index in RequireSetup

File: SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Common/LinuxScripts.cs (hunk 1 above)
Comment (davedolben) [reply SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Common/LinuxScripts.cs]: Done in all the build cs files

Comment (davedolben) [reply SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Common/LinuxScripts.cs]: Ditto

**Comment by m-samiec** [comment]
Ran locally and built a project to verify - all looks good.
